pacman/operations/algorithm_reports/reports_metadata.xml

```diff
@@ -116,6 +116,34 @@
             <param_name>machine</param_name>
         </required_inputs>
     </algorithm>
+    <algorithm name="RouterSummaryReport">
+        <python_module>pacman.operations.algorithm_reports.reports</python_module>
+        <python_function>router_summary_report</python_function>
+        <input_definitions>
+            <parameter>
+                <param_name>report_folder</param_name>
+                <param_type>ReportFolder</param_type>
+            </parameter>
+            <parameter>
+                <param_name>routing_tables</param_name>
+                <param_type>MemoryRoutingTables</param_type>
+            </parameter>
+            <parameter>
+                <param_name>hostname</param_name>
+                <param_type>IPAddress</param_type>
+            </parameter>
+            <parameter>
+                <param_name>machine</param_name>
+                <param_type>MemoryExtendedMachine</param_type>
```

Comment:
what about MemoryMachine? extended (if memory serves) is only built with external devices been added. what about a setup which never had to add any?

Reply:
The chip allocater always runs, so you always get a MemoryExtendedMachine at this point I believe.  You can of course use MemoryMachine if you are not going to look at virtual chips in any case, since they are both the same in terms of real chips.

Reply:
GIGO

Reply:
so can have a second type of MemoryMachine (and it might even be better as just a MemoryMachine, as rowley hath pointed out). so not GIGO.  😛

Reply:
copied directly from routingInfoReports   so if this wrong it is because existing code was wrong so GIGO!

Reply:
so your telling me that coz the foundations is wrong, your just going to keep building on the shoddy foundations? remind me never to go to a house you built. 😜